mlir/lib/Dialect/Linalg/Transforms/Vectorization.cpp

```diff
@@ -2488,10 +2542,11 @@ vectorizeScalableVectorPrecondition(Operation *op,
 
   auto linalgOp = dyn_cast<LinalgOp>(op);
 
-  // Cond 1: There's been no need for scalable vectorisation of
-  // non-linalg Ops so far
-  if (!linalgOp)
-    return failure();
+  // Cond 1: Reject Ops that don't implement the LinalgOp interface, with the
+  // exception of UnpackOp for which there is a dedicated hook.
+  if (!linalgOp) {
+    return isa<linalg::UnPackOp>(op) ? success() : failure();
+  }
```

Comment:
Please also update the function comment. Now it accepts linalg.unpack ops.

Is the below form simpler to you?

```cpp
  if (!linalgOp)
    return success(isa<linalg::UnPackOp>(op));
```

https://github.com/llvm/llvm-project/blob/56ae79a6ab70ded92846e053b13745ea51e48089/llvm/include/llvm/Support/LogicalResult.h#L27-L31

Reply:
> Is the below form simpler to you?

Yes, thanks for the suggestion 🙏🏻